src/sas/sascalc/dataloader/file_reader_base_class.py

```diff
@@ -80,8 +80,14 @@ def read(self, filepath):
             if self.extension in self.ext or self.allow_all:
                 # Try to load the file, but raise an error if unable to.
                 try:
-                    self.f_open = open(filepath, 'rb')
-                    self.get_file_contents()
+                    try:
+                        self.f_open = open(filepath, 'rb')
+                        self.get_file_contents()
+                    except ValueError:
+                        # Issue(s) found while reading file in binary mode
+                        # Attempt to open in text-only mode
+                        self.f_open = open(filepath, 'r')
+                        self.get_file_contents()
```

Comment:
If `get_file_contents` fails on `'rb'` then the file isn't closed automatically, but instead relies on the garbage collector to close it.  Better to use a context manager:
```python
try:
    with open(filepath, 'rb) as self.f_open:
        self.get_file_contents()
except ValueError:
    with open(filepath, 'r') as self.f_open:
        self.get_file_contents()
```

I don't think you need the `finally` block if you do this. The file will be closed by the context manager, and the other operations (the extension deprecation method and the data sorting) didn't ever need to be in `finally`. Since the only exception bypass is for the `FileContentsException`, as well as external control exceptions like the keyboard interrupt and the quit signal, the other operations can occur after the exception block is complete.

Probably too much for this PR, but it would be better to avoid adding temporary state to the object. Can `get_file_content` be modified to accept a file handle? Then you can use:
```
with open(filepath, 'rb') as f_open:
    self.get_file_contents(f_open)
```

Reply:
I have implemented the `with open()` changes on my local machine. I did not make the suggested change to pass f_open to get_file_contents - that would require changes in almost every reader.

Fixing the ANSI file loading issue in Linux (works in Windows) is trickier, so I am closing this PR until I have a solution.